### ISSUE TITLE
[Bug]: different result in mo and mysql with sql 'select UNIX_TIMESTAMP(date)'.

### ISSUE THREAD

**Issue by Ariznawlll** [matrixorigin/matrixone#14804]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Branch Name

main

### Commit ID

868980a8664bcac9f8d268e0467a720cc810070b

### Other Environment Information

```Markdown
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

<img width="1436" alt="image" src="https://github.com/matrixorigin/matrixone/assets/108530700/d1621ca5-58cf-40b3-b79d-312cdd788204">

### Expected Behavior

<img width="1436" alt="image" src="https://github.com/matrixorigin/matrixone/assets/108530700/e52b15ca-e376-417b-9ad3-823f5f2625b5">

### Steps to Reproduce

```Markdown
drop table if exists time01;
create table time01(col1 date, col2 datetime, col3 timestamp, col4 time);
insert into time01 values ('2020-10-11', '2023-11-11 10:00:01', '1997-01-13 12:12:12.000', '12:12:12');
insert into time01 values ('1919-12-01', '1990-10-10 01:01:01', '2001-12-12 01:01:01.000', '10:59:59');
insert into time01 values (null, null, null, null);
select * from time01;

select unix_timestamp(col1) from time01;
```

**Comment by fengttt** [comment]
I don't think it is a bug.   MySQL result is definitely wrong, I am not sure if NULL is the right thing, but it is better than 0.

**Comment by matrix-meow** [comment]
Hello @Ariznawlll. The bug issue in the BVT test code has not been removed,issues automatically open.

**Comment by aunjgr** [comment]
PR合并之后可以关掉
